server/server.js

```diff
@@ -1,124 +1,128 @@
-require('dotenv').config()
-
 global.XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest
 
 const express = require('express')
 const path = require('path')
 const webpack = require('webpack')
 const cors = require('cors')
-const Spotify = require('spotify-web-api-js')
 const webpackDevMiddleware = require('webpack-dev-middleware')
 const config = require('../webpack.config')
-const refreshAccessToken = require('./spotify-connector')
 
-const port = process.env.PORT
 const compiler = webpack(config)
-const app = express()
-const spotifyApi = new Spotify()
-
-app.use(cors())
-app.options('*', cors())
-// app.use(webpackDevMiddleware(compiler, {
-//   publicPath: config.output.publicPath
-// }))
-
-app.use(express.static(config.output.publicPath))
-
-refreshAccessToken() // Returns accessToken
 
-function errorHandler(err) {
-  if (err.status === 401) {
-    console.log('token refreshed')
-    refreshAccessToken()
-  } else {
-    console.log(err)
+class Server {
+  constructor(Config, SpotifyConnector) {
+    this.Config = Config
+    this.SpotifyConnector = SpotifyConnector
   }
-}
-
-app.get('/api/artist/:name', (req, res) => {
-  spotifyApi.searchArtists(`${req.params.name}*`, {
-    limit: 5
-  }, (err, data) => {
-    if (err) {
-      errorHandler(err)
+      
+  errorHandler(err) {
+    if (err.status === 401) {
+      console.log('token refreshed')
+      this.SpotifyConnector.refreshAccessToken()
     } else {
-      res.send(data)
+      console.log(err)
     }
-  })
-})
+  }
 
-app.get('/api/findartist/:name', (req, res) => {
-  spotifyApi.getArtist(req.params.name,
-    (err, data) => {
-      if (err) {
-        errorHandler(err)
-      } else {
-        res.send(data)
-      }
+  initServer() {
+    this.app = express()
+    this.app.use(cors())
+    this.app.options('*', cors())
+    // app.use(webpackDevMiddleware(compiler, {
+    //   publicPath: config.output.publicPath
+    // }))
+    
+    this.app.use(express.static(config.output.publicPath))
+    this.SpotifyConnector.refreshAccessToken()
+    
+    this.app.get('/api/artist/:name', (req, res) => {
+      this.SpotifyConnector.SpotifyApi.searchArtists(`${req.params.name}*`, {
```

Comment:
Hmm trochę pomieszane. Teraz nie do końca wiadome jakie zadanie ma SpotifyConnector. Jeżeli ma służyć tylko do połączenia z SpotifyApi to imho powinno być coś takiego:
```
// Metoda zwraca `SpotifyApi`
this.api = this.SpotifyConnector.connect()

// I wtedy używasz już api bezpośrednio
this.api.searchArists()
```

A jeżeli chcesz, żeby to był wrapper wokół API to powinno to wyglądać tak I guess:
```
const { data, err } = this.SpotifyConnector.searchArtists(reg.params.name)
if (err) {...}
else res.send(data)
```

Reply:
dodalem gettera, ktory zwraca spotifyApi i uzywam go w constructorze servera teraz